allennlp/nlp_api.py

```diff
@@ -0,0 +1,289 @@
+from typing import Callable, Dict, List, Union
+
+import torch
+
+from .common.checks import ConfigurationError
+from .common import Params
+
+
+class NlpApi:
+    """
+    The ``NlpApi`` provides a high-level API for building NLP models.  You don't have to use it
+    for building a model with AllenNLP, but we think you should, because it's useful.
+
+    ``NlpApi`` abstracts away particular decisions that are frequently necessary in NLP, like how
```

Comment:
My understanding of the encouraged pattern is that client code/config should look like:
```
model.py
nlp.get_recurrent_cell(name="my_rnn")

experiment.json
"my_rnn" = "seq2seq"
```

Would be nice to have a simple concrete [and precise!] example right here in the overview docstring.